src/Player.cpp

```diff
@@ -2264,34 +2270,64 @@ void Player::Step( int col, int row, const RageTimer &tm, bool bHeld, bool bRele
 				// bug. (It was fNoteOffset > 0.f before) -DaisuMaster
 				if( !REQUIRE_STEP_ON_HOLD_HEADS && ( fNoteOffset <= GetWindowSeconds( TW_W5 ) && GetWindowSeconds( TW_W5 ) != 0 ) )
 				{
-					score = TNS_W1;
+					// Set it to the first non-disabled window.
+					const auto &disabledWindows = m_pPlayerState->m_PlayerOptions.GetCurrent().m_twDisabledWindows;
+					if (disabledWindows.find(TW_W1) == disabledWindows.end())
```

Comment:
That would be `if (!disabledWindows[TW_W1])` with a bitset ;)

Reply:
Done!